kythe/typescript/indexer.ts

```diff
@@ -1139,15 +1139,39 @@ class Visitor {
           const kParent = this.getSymbolName(parentSym, TSNamespace.TYPE);
           this.emitEdge(kFunc, 'childof', kParent);
         }
-      }
 
-      // TODO: emit an "overrides" edge if this method overrides.
-      // It appears the TS API doesn't make finding that information easy,
-      // perhaps because it mostly cares about whether types are structrually
-      // compatible.  But I think you can start from the parent class/iface,
-      // then from there follow the "implements"/"extends" chain to other
-      // classes/ifaces, and then from there look for methods with matching
-      // names.
+        // Emit "overrides" edges if this method overrides extended classes or
+        // implemented interfaces, which are listed in the Heritage Clauses of
+        // a class or interface.
+        //     class X extends A implements B, C {}
+        //             ^^^^^^^^^-^^^^^^^^^^^^^^^----- `HeritageClause`s
+        // Look at each type listed in the heritage clauses and traverse its
+        // members. If the type has a member that matches the method visited in
+        // this function (`kFunc`), emit an "overrides" edge to that member.
+        if (funcSym && decl.parent.heritageClauses) {
+          for (const heritage of decl.parent.heritageClauses) {
+            for (const baseType of heritage.types) {
+              const baseSym = this.getSymbolAtLocation(baseType.expression);
```

Comment:
One case where this may happen is 

```
class X extends Array<string> {}
```

I forget whether generics get their own symbol or not.

I wonder whether it's worth marking this as extending just "Array" or something in that case?  But I think that also will do a very wrong thing in the case where it's some sort of mapped type, so maybe it's better to just leave that alone for now.

Reply:
Generics don't get their own symbol. This would emit an overrides edge to something in Array I think.